[docs] Update Firebase JS SDK config steps (#36641)
expo/expo: docs/pages/guides/using-firebase.mdx

Expo CLI uses [Metro](https://metrobundler.dev/) to bundle your JavaScript code and assets, and add [support for more file extensions](/guides/customizing-metro/#adding-more-file-extensions-to--assetexts).

Start by generating the template file **metro.config.js** in your project's root directory using the following command:

<Terminal cmd={['$ npx expo customize metro.config.js']} />

Firebase JS SDK loads both ES modules (ESM) and CommonJS modules (CJS) when bundled in your project. To work around this, ensure Metro is configured to support ESM from the Firebase JS SDK using the following configuration:

```js metro.config.js
const { getDefaultConfig } = require('@expo/metro-config');

/** @type {import('expo/metro-config').MetroConfig} */
const config = getDefaultConfig(__dirname);

config.resolver.resolveRequest = (context, moduleImport, platform) => {
  // Always import the ESM version of all `@firebase/*` packages
  if (moduleImport.startsWith('@firebase/')) {
    return context.resolveRequest(
      {
        ...context,
        isESMImport: true,
      },
      moduleImport,
      platform
    );
  }

  return context.resolveRequest(context, moduleImport, platform);
};

> module.exports = defaultConfig;

Author: ```suggestion
module.exports = config;
```

</Step>

<Step label="2">
#### Configure Metro

Expo CLI uses [Metro](https://metrobundler.dev/) to bundle your JavaScript code and assets, and add [support for more file extensions](/guides/customizing-metro/#adding-more-file-extensions-to--assetexts).

Start by generating the template file **metro.config.js** in your project's root directory using the following command:

<Terminal cmd={['$ npx expo customize metro.config.js']} />

> Firebase JS SDK library loads both ES modules (ESM) and Common JS modules (CJS) when bundled in your project. To work around this, ensure Metro is configured to support ESM or CJS modules but not both. The following configuration ensures that Metro uses ESM format for the Firebase JS SDK:

Author: ```suggestion
Firebase JS SDK loads both ES modules (ESM) and CommonJS modules (CJS) when bundled in your project. To work around this, ensure Metro is configured to support ESM from the Firebase JS SDK using the following configuration:
```
